### ISSUE TITLE
Provide point markers to both ends of a rule if specified

### ISSUE THREAD

**Issue by jwoLondon** [vega/vega-lite#7259]
Specifying a point marker for a rule almost allows a simple and intuitive specification of a dumbbell chart (ranged dotplot):

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  "data": {
    "values": [
      { "sample": 1, "mins": 10, "maxs": 23 },
      { "sample": 2, "mins": 14, "maxs": 20 },
      { "sample": 3, "mins": 17, "maxs": 31 }
    ]
  },
  "encoding": {
    "x": { "field": "sample" },
    "y": { "field": "mins", "type": "quantitative" },
    "y2": { "field": "maxs" }
  },
  "mark": {"type":"rule", "point": true}
}
```

However, currently it only places a marker at the `y` end of each rule with a warning about the point marker being incompatible with `y2`. It would be nice if markers were placed at both ends when `x`/`x2` or `y`/`y2` pairs are specified.

Currently this also interacts with #7257 in that replacing the rule mark in the example above with a line mark will produce the same output, but removing `"point": true` will then no longer convert the line to a rule.

**Comment by domoritz** [comment]
I agree this would be great. This should be implemented in the normalized (similar to how we implement error bars and boxplots).
